### PR TITLE
feat(plutil): Added completion spec for plutil

### PR DESCRIPTION
**Pull request by duncanlutz**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature

**What is the current behavior? (You can also link to an open issue here)**
No completion spec for plutil currently exists ([issue that requested this feature](https://github.com/withfig/autocomplete/issues/1541))

**What is the new behavior (if this is a feature change)?**
Added completion spec for plutil

### CODE REVIEW

File: src/plutil.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  name: "plutil",
+  description:
+    "Check the syntax of property list files, or convert a plist file from one format to another. Specifying - as an input file reads from stdin",
+  parserDirectives: {
+    flagsArePosixNoncompliant: true,
+    optionsMustPrecedeArguments: true,
+  },
+  args: {
+    name: "file",
+    description: "File to check the syntax of or convert",
```
Comment (fedeci) [inline src/plutil.ts:11]:
```suggestion
    description: "File to check the syntax of or convert",
    template: "filepaths",
```

File: src/plutil.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      name: "-r",
+      description:
+        "For JSON, add whitespace and indentation to make the output more human-readable",
+    },
+    {
+      name: "-o",
+      description:
+        "Specify an alternate path name for the result of the -convert operation; this option is only useful with a single file to be converted. Specifying - as the path outputs to stdout",
+      args: {
+        name: "path",
```
Comment (fedeci) [inline src/plutil.ts:68]:
```suggestion
        name: "path",
        template: "filepaths",
        suggestCurrentToken: true,
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks! Just two comments about some improvements that can be done to the spec!

**Review by SeparateRecords — CHANGES_REQUESTED** [review]
Thanks @duncanlutz! Got some comments on the general structure

Some of `plutil`'s options aren't _really_ options. It takes some "command options" which we can model as subcommands.

Some subcommands are `-convert`, `-extract`, etc.

This also allows us to model the behavior more closely too

```ts
  subcommands: [
    {
      name: "-extract",
      icon: "fig://icon?type=option", // makes it look like an option
      args: [
        { name: "keypath" },
        { name: "format", suggestions: ["xml1", "binary1", "json", "swift", "objc", "raw"] },
        { name: "file", template: "filepaths", suggestions: [{ name: "-", hidden: true }] },
      ],
      options: [
        {
          name: "-expect",
          args: {
            name: "type",
            suggestions: ["-bool", "-integer", "-float", /* ... */ ],
          },
        },
      ],
    }
  ]
```

**Review by mschrage — APPROVED** [review]

**Pull request merged**